### PR TITLE
feature(User): Add List starred gists API

### PR DESCRIPTION
**Pull request by knsh14**
https://developer.github.com/v3/gists/#list-starred-gists

### CODE REVIEW

File: lib/User.js [hunk 1]
```diff
@@ -132,6 +132,23 @@ class User extends Requestable {
...
+    * @return {Promise} - the promise for the http request
+    */
+   listStarredGists(options, cb) {
+      options = options || {};
+      if (typeof options === 'function') {
+         cb = options;
+         options = {};
+      }
+      options.since = this._dateToISO(options.since);
+      return this._request('GET', "/gists/starred", options, cb);
```
Comment (j-rewerts) [inline lib/User.js]: Nit: switch to single quotes.

**Comment by j-rewerts** [comment]
After fixing the lint issue, we can merge this in.

**Comment by knsh14** [comment]
thanks for comment!
just fixed your review point.

**Comment by j-rewerts** [comment]
@clayreimann Feel free to merge.

**Pull request merged**